File: INTRODUCCION-POO/EJERCICIO 3/Coche.java
package Ejercicio_3;

public class Coche {
	public String marca;
    public String modelo;
    public int velocidad;

    public Coche(String marca, String modelo) {
        this.marca = marca;
        this.modelo = modelo;
        this.velocidad = 20;
    }

    public void acelerar() {
        this.velocidad += 15;
    }

    public void frenar() {
        this.velocidad -= 5;
        if (this.velocidad < 0) {
            this.velocidad = 0;
        }
    }

    public String toString() {
        return "Coche: " + this.marca + " " + this.modelo + ", Velocidad: " + this.velocidad + " km/h";
    }
}

File: INTRODUCCION-POO/EJERCICIO 3/Coche.py
class Coche:
    def __init__(self, marca, modelo):
        self.marca = marca
        self.modelo = modelo
        self.velocidad = 0

    def acelerar(self):
        self.velocidad += 10

    def frenar(self):
        self.velocidad -= 5
        if self.velocidad < 0:
            self.velocidad = 0

    def __str__(self):
        return f'Coche {self.marca} {self.modelo}, Velocidad: {self.velocidad} km/h'


c1 = Coche("Toyota", "Corolla")
c2 = Coche("Honda", "Civic")
c1.acelerar()
c1.acelerar()
c1.frenar()

c2.acelerar()
c2.frenar()
c2.frenar()

print(c1)
print(c2)


